android-o/AutofillFramework/AutofillFramework/multidatasetservice/model/FilledAutofillFieldCollection.cs

```diff
@@ -0,0 +1,142 @@
+		{
+			bool setValueAtLeastOnce = false;
+			List<string> allHints = autofillFieldMetadataCollection.AllAutofillHints;
+			for (int hintIndex = 0; hintIndex < allHints.Count; hintIndex++)
+			{
+				string hint = allHints[hintIndex];
+				List<AutofillFieldMetadata> fillableAutofillFields = autofillFieldMetadataCollection.GetFieldsForHint(hint);
+				if (fillableAutofillFields == null)
+				{
+					continue;
+				}
+				for (int autofillFieldIndex = 0; autofillFieldIndex < fillableAutofillFields.Count; autofillFieldIndex++)
+				{
+					FilledAutofillField filledAutofillField = HintMap[hint];
+					if (filledAutofillField == null)
+					{
+						continue;
+					}
+					AutofillFieldMetadata autofillFieldMetadata = fillableAutofillFields[autofillFieldIndex];
+					var autofillId = autofillFieldMetadata.AutofillId;
+					var autofillType = autofillFieldMetadata.AutofillType;
+					switch (autofillType)
+					{
+						case (int) AutofillType.List:
+							var listValue = autofillFieldMetadata.GetAutofillOptionIndex(filledAutofillField.TextValue);
+							if (listValue != -1)
+							{
+								datasetBuilder.SetValue(autofillId, AutofillValue.ForList(listValue));
+								setValueAtLeastOnce = true;
+							}
+							break;
+						case (int) AutofillType.Date:
+							var dateValue = filledAutofillField.DateValue;
+							datasetBuilder.SetValue(autofillId, AutofillValue.ForDate((long)dateValue));
+							setValueAtLeastOnce = true;
+							break;
+						case (int) AutofillType.Text:
+							var textValue = filledAutofillField.TextValue;
+							if (textValue != null)
+							{
+								datasetBuilder.SetValue(autofillId, AutofillValue.ForText(textValue));
+								setValueAtLeastOnce = true;
+							}
+							break;
+						case (int) AutofillType.Toggle:
+							var toggleValue = filledAutofillField.ToggleValue;
+							if (toggleValue != null)
+							{
+								datasetBuilder.SetValue(autofillId, AutofillValue.ForToggle(toggleValue.Value));
+								setValueAtLeastOnce = true;
+							}
+							break;
+						default:
+							Log.Warn(CommonUtil.Tag, "Invalid autofill type - " + autofillType);
+							break;
+					}
+				}
+			}
+			return setValueAtLeastOnce;
+		}
+
+		/// <summary>
+		/// Takes in a list of autofill hints (`autofillHints`), usually associated with a View or set of
+		/// Views. Returns whether any of the filled fields on the page have at least 1 of these
+		/// `autofillHint`s.
+		/// </summary>
+		/// <returns><c>true</c>, if with hints was helpsed, <c>false</c> otherwise.</returns>
+		/// <param name="autofillHints">Autofill hints.</param>
+		public bool HelpsWithHints(List<string> autofillHints)
+		{
+			for (int i = 0; i < autofillHints.Count; i++)
+			{
+				var autofillHint = autofillHints[i];
+				if (HintMap.ContainsKey(autofillHint) && !HintMap[autofillHint].IsNull())
+				{
+					return true;
+				}
+			}
+			return false;
+		}
+
+		public static implicit operator FilledAutofillFieldCollection(Java.Lang.Object v) {
```

Comment:
I may be missing something but I'm not exactly sure how this is being used? Perhaps we can remove it?

Reply:
I don't remember why I add this. I think there was a problem casting some types and I had to implement it to allow the assignment. I've removed this and it's working fine